x/stake/client/cli/query.go

```diff
@@ -15,63 +15,63 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/stake"
 )
 
-// get the command to query a candidate
-func GetCmdQueryCandidate(storeName string, cdc *wire.Codec) *cobra.Command {
+// get the command to query a validator
+func GetCmdQueryValidator(storeName string, cdc *wire.Codec) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "candidate",
-		Short: "Query a validator-candidate account",
+		Use:   "validator [owner-addr]",
+		Short: "Query a validator",
+		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 
-			addr, err := sdk.GetAddress(viper.GetString(FlagAddressCandidate))
+			addr, err := sdk.GetAddress(args[0])
 			if err != nil {
 				return err
 			}
-
-			key := stake.GetCandidateKey(addr)
+			key := stake.GetValidatorKey(addr)
 			ctx := context.NewCoreContextFromViper()
 			res, err := ctx.Query(key, storeName)
 			if err != nil {
 				return err
 			}
 
-			// parse out the candidate
-			candidate := new(stake.Candidate)
-			cdc.MustUnmarshalBinary(res, candidate)
-			output, err := wire.MarshalJSONIndent(cdc, candidate)
+			// parse out the validator
+			validator := new(stake.Validator)
+			cdc.MustUnmarshalBinary(res, validator)
```

Comment:
Should probably get rid of the MustUnmarshal and just use the one that returns err

Reply:
I disagree, the client is retrieving from the store - it's unexpected behaviour that is would ever panic here

Reply:
fair enough - we certainly dont need both tho

Reply:
oh lawl! didn't see the second one in the comment view box. - thanks for the catch